### PR TITLE
Misc changes for GC in new MM

### PR DESCRIPTION
**Pull request by projedi**

### CODE REVIEW

File: backend.native/tests/build.gradle [hunk 1]
```diff
@@ -2396,6 +2396,11 @@ task hash_map0(type: KonanLocalTest) {
     source = "runtime/collections/hash_map0.kt"
 }
 
+task hash_map1(type: KonanLocalTest) {
+    goldValue = "OK\n"
+    source = "runtime/collections/hash_map1.kt"
```
Comment (projedi) [inline backend.native/tests/build.gradle:2401]: Extracted long tests here.

File: backend.native/tests/runtime/collections/hash_map1.kt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        for (i in 1..n) {
+            assertFalse(m.containsKey(i.toString()))
+        }
+    }
+}
+
+@Test fun runTest() {
+    // TODO: Do not manually control this.
+    if (Platform.memoryModel == MemoryModel.EXPERIMENTAL) {
+        GC.threshold = 1000000
```
Comment (projedi) [inline backend.native/tests/runtime/collections/hash_map1.kt:94]: In these tests there're a lot of safepoints, so until we have a proper GC scheduling mechanism, we have to manually tune thresholds here.

File: runtime/src/mm/cpp/GlobalData.hpp [hunk 3]
```diff
@@ -29,8 +29,9 @@ class GlobalData : private Pinned {
 
 private:
     GlobalData();
-    ~GlobalData();
+    ~GlobalData() = delete;
 
+    // This `GlobalData` is never destroyed.
```
Comment (projedi) [inline runtime/src/mm/cpp/GlobalData.hpp:34]: With the new MM we don't shutdown the runtime at the program finish, so no need to destroy `GlobalData` too.

File: runtime/src/mm/cpp/InitializationSchemeTest.cpp [hunk 4]
```diff
@@ -39,6 +39,7 @@ class InitSingletonTest : public testing::Test {
         // Make sure to clean everything allocated by the tests.
         for (auto& threadData : threadDatas_) {
             threadData->objectFactoryThreadQueue().ClearForTests();
+            threadData->globalsThreadQueue().ClearForTests();
```
Comment (projedi) [inline runtime/src/mm/cpp/InitializationSchemeTest.cpp:42]: `globalsThreadQueue()` will publish into a global (inside `GlobalData`) when `threadData` is destroyed, so we need to clean them up before it happens. In the future I'll look into decoupling tests from accessing globals.

File: runtime/src/mm/cpp/Memory.cpp [hunk 5]
```diff
@@ -263,6 +263,48 @@ extern "C" void Kotlin_native_internal_GC_collectCyclic(ObjHeader*) {
     ThrowIllegalArgumentException();
 }
 
+extern "C" void Kotlin_native_internal_GC_setThreshold(ObjHeader*, int32_t value) {
```
Comment (projedi) [inline runtime/src/mm/cpp/Memory.cpp:266]: Reusing existing `kotlin.native.internal.GC` API for tuning GC. It's a temporary solution.

**Review by ilmat192 — APPROVED** [review]

**Pull request merged**